pkg/plugins/policies/meshaccesslog/api/v1alpha1/meshaccesslog.go

```diff
@@ -51,24 +49,24 @@ type Backend struct {
 type TCPBackend struct {
 	// Format of access logs. Placeholders available on
 	// https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log
-	Format Format `json:"format,omitempty"`
+	Format *Format `json:"format,omitempty"`
 	// Address of the TCP logging backend
-	Address string `json:"address,omitempty"`
+	Address string `json:"address"`
 }
 
 // FileBackend defines configuration for file based access logs
 type FileBackend struct {
 	// Format of access logs. Placeholders available on
 	// https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log
-	Format Format `json:"format,omitempty"`
+	Format *Format `json:"format,omitempty"`
 	// Path to a file that logs will be written to
-	Path string `json:"path,omitempty"`
+	Path string `json:"path"`
 }
 
 type Format struct {
-	Plain           string      `json:"plain,omitempty"`
-	Json            []JsonValue `json:"json,omitempty"`
-	OmitEmptyValues bool        `json:"omitEmptyValues,omitempty"`
+	Plain           *string      `json:"plain,omitempty"`
+	Json            *[]JsonValue `json:"json,omitempty"`
```

Comment:
slice can already be nil, do we need a pointer here?

Reply:
https://go.dev/play/p/5gAykZVuEAS

Reply:
Finally somebody asks the big question about slices! 

I suspect that the generation code as is expects a pointer type.

IMO we should use a pointer if we can, as long as we really treat `xs := nil` just like `xs := []X{}` everywhere. I think we just have to be careful/make sure that we never serialize an above value to `{ json: null }`, for example what happens with:

```
var values []JsonValue
f := Format{ Json: &values }
```


? I'm not positive.

Reply:
https://go.dev/play/p/Nfpf30MCi0J indicates that this is dangerous

Reply:
> IMO we should use a pointer if we can

But why, what is the advantage?
Because I can see a disadvantage of writing extra code that has to deal with nils. Also, it's easier to get panic on nil dereference, because Go has no mechanism to protect us from it (except default values if you drop pointers).

Reply:
The advantage in the abstract is of course that the semantics of a non-pointer slice would be the same as any non-pointer value. It's absolutely insane to create a type where there are 2 different "empty lists", aka values of length 0 but that's what go did.

> Because I can see a disadvantage of writing extra code that has to deal with nils

That's what I meant, if we can, given go as is. So if the danger of serializing a nil slice as `null` exists, we can't use a pointer wrapper.

> Also, it's easier to get panic on nil dereference, because Go has no mechanism to protect us from it (except default values if you drop pointers).

Isn't the slice case the only one where we _can't_ get a panic because of a nil dereference? I.e. we have to deal with it for every other type.

Reply:
If you have:
```go
type Spec struct {
    MySlice []MyType `json:"mySlice,omitempty"`
}
```
then when you serialize to json it won't distinguish between nil and empty slice. Both will be serialized to:
```json
"spec": {}
```
There is a proposal to add `omitnil` https://github.com/golang/go/issues/22480, but it doesn't seem like it's going to happen

Reply:
That's bad, right? Because we need to have a value that means "not set". So the only way to express an optional slice, without the danger of empty slice vs nil slice, would be 

```
    MySlice []MyType `json:"mySlice"`
```

Reply:
Maybe there's a better JSON encoding lib for go

Reply:
Yes, but you also need annotations:
```
// +optional
// +nullable
 MySlice []MyType `json:"mySlice"`
 ```
 and in that case, it will always serialize omitted by user values to `mySlice: null`

Reply:
That's actually what we had in MeshAccessLog before this PR https://github.com/kumahq/kuma/pull/5580/files#diff-ed8e7e0fddb480be4d601ae35bb953cddd53659fb3b9d306b43996a8f322cb38L40

Reply:
So I don't care about which annotations we have to use but what I want ideally is

```
type Struct struct {
  MySlice *[]MyType
}
```

where a `nil` pointer means not set, just like for every other possible field type. Where a nil slice and an empty slice are the exact same thing. So I can just use `val.MySlice == nil` to check whether the value is provided and `len(val.MySlice) == 0` to check whether an empty slice has been set.

and where encoded values are either `{}`, `{mySlice: []}`, or `{mySlice: [...]}` and there is no `{mySlice: null}`

Reply:
> I don't know if it'd violate our generated open API specs

Yes, If I recall correctly it violates the specs without `// +nullable`

Reply:
Ok